src/App.tsx

```diff
@@ -15,36 +15,90 @@
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+export const App = () => {
+  const [sortType, setSortType] = useState('default');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getSortedGoods = () => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortType === 'alphabet') {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortType === 'length') {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const handleSortAlphabetically = () => {
+    setSortType('alphabet');
+  };
+
+  const handleSortByLength = () => {
+    setSortType('length');
+  };
+
+  const handleReverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const handleReset = () => {
+    setSortType('default');
+    setIsReversed(false);
+  };
+
+  const isActive = sortType !== 'default' || isReversed;
+  const sortedGoods = getSortedGoods();
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'alphabet' ? '' : 'is-light'}`}
+          onClick={handleSortAlphabetically}
```

Comment:
[UX] - To mark the active button, ensure that the button visually indicates its active state. The current implementation does not change the appearance of the button when it is active. Consider adding a distinct style or class when `sortType === 'alphabet'`.